crates/aptos/src/move_tool/lint.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use super::IncludedArtifactsArgs;
+use crate::common::types::{CliCommand, CliTypedResult, MovePackageDir, OptimizationLevel};
+use aptos_framework::{BuildOptions, BuiltPackage};
+use async_trait::async_trait;
+use clap::Parser;
+use move_compiler_v2::Experiment;
+use move_model::metadata::{CompilerVersion, LanguageVersion};
+
+#[derive(Parser)]
+pub struct LintPackage {
+    // TODO: add some options to select certain lint/warning passes?
+    #[clap(flatten)]
+    pub(crate) included_artifacts_args: IncludedArtifactsArgs,
+
+    #[clap(flatten)]
+    pub(crate) move_options: MovePackageDir,
+}
+
+#[async_trait]
+impl CliCommand<&'static str> for LintPackage {
+    fn command_name(&self) -> &'static str {
+        "LintPackage"
+    }
+
+    async fn execute(self) -> CliTypedResult<&'static str> {
+        let move_options = MovePackageDir {
+            lint: true,
+            optimization_level: Some(OptimizationLevel::None),
+            move_2: true,
+            // TODO: These should be set more automatically.
+            language_version: Some(LanguageVersion::V2_0),
+            compiler_version: Some(CompilerVersion::V2_0),
+            bytecode_version: Some(7),
+            ..self.move_options.clone()
+        };
+        if matches!(
+            self.move_options.language_version,
+            Some(LanguageVersion::V1)
+        ) {
+            eprintln!("Note that `aptos move lint` requires Move Language Version 2 and above");
+            static EINVAL: i32 = 22;
+            std::process::exit(EINVAL);
```

Comment:
This follows a non-standard process of error handling

Reply:
Point me at the standard process, please?

Reply:
Ok, I'll just panic.  That seems the usual way.

Reply:
Definitely panic is not the way how the CLI reports errors.

Reply:
Please return `CliResult<>`, which will take the error and provide the error as a JSON output, similar to all other outputs.

Reply:
Done already.